src/crimson/admin/admin_socket.cc

```diff
@@ -88,6 +88,9 @@ auto AdminSocket::parse_cmd(const std::vector<std::string>& cmd)
   try {
     cmd_getval(cmdmap, "format", format);
     cmd_getval(cmdmap, "prefix", prefix);
+    if (prefix == "pg"){
+      cmd_getval(cmdmap, "cmd", prefix);
+    }
```

Comment:
This doesn't look right -- admin_socket.cc probably shouldn't be encoding osd specific rules.  What behavior is this trying to fix?  @ronen-fr I'll add you as a reviewer.

Reply:
crimson not use pg as prefix, it use pg cmd such as "query" as prefix.

Reply:
can't us pg prefix to find registered hook function.

Reply:
@ronen-fr ^

Reply:
@ronen-fr could you please take a look on it?

Reply:
after Sage's overhaul of the tell command back in #30217 , there are two forms of asock backed command when it comes "pg" commands

- the old one: `ceph pg <pgid> <cmd>`. where `<cmd>` can be `{query,mark_unfound_lost,list_unfound,scrub,deep_scrub}`. so far, we do have a dedicated `PGCommand` for handling pg commands. it is able to consolidate the pg specific logic. but it fails to register the `<cmd>` under the `"pg"` prefix. while in classic OSD, `PrimaryLogPG::do_command()` simply assign the value of `"cmd"` to that of `"prefix"` if the `"prefix"` is `"pg"`.
- the new one: `ceph tell <pgid> <cmd>`. the idea is that this would be consistent. as we also have commands like `ceph tell <osd-id> <cmd>` and `ceph tell <mon-id> <cmd>`, etc. 

but unfortunately, quite a few of our tests are still using the old form. i think there are three possible solution to address this problem:

- just blindly set prefix to the value of `"cmd"` if the prefix is `"pg"`. this practically replicates the hack in `PrimaryLogPG::do_command()`.
- fix all tests so they use the new forms. but that'd be a more aggressive change. and add a dedicated `PgCommandsHook` which just print a warning message explaining that this command is deprecated.
- add a dedicated `PgCommandsHook`, which delegates the call back to `AdminSocket`. so we need to:
  1. extract a new member function out of `AdminSocket::execute_command()`,   so we can reuse the new function to call a command with `prefix: string`, `cmdmap: cmdmap_t`, `format: string`, and `buf: ceph::bufferlist`.
  2. add a dedicated member variable `std::set pg_commands` in `AdminSocket` for tracking pg related commands
  3. specialize `AdminSocket::register_command()` and `make_asok_hook()`, so the former can add the `PGCommand` commands' prefixes into the `pg_commands `
  4. add `PgCommandsHook` with the prefix of `"pg"`. if the `"cmd"` param is in `pg_commands`, it should pass the parameters like `cmdmap` down to the extracted member function.

Reply:
@athanatos what is your opinion?  I think this fix just as: 
>>just blindly set prefix to the value of "cmd" if the prefix is "pg". this practically replicates the hack in PrimaryLogPG::do_command().

Reply:
Hmm, ok.  Add a comment:

```
// tolerate old-style pg <pgid> command <args> style formatting
```